test/test_nn.py

```diff
@@ -10498,6 +10498,17 @@ def test_softmax_double(self, device, dtype):
         expected_ones = F.log_softmax(logits, dim=1).exp().sum(dim=1)
         self.assertEqual(expected_ones, torch.ones_like(expected_ones))
 
+        # backward
+        logits = torch.randn(5, 513, dtype=dtype, device=device, requires_grad=True)
```

Comment:
does it actually go through that kernel?

Reply:
it actually goes through the kernel but there's a heuristic determining the block size rather than hardcoding it to 512 and I couldn't find a case that actually causes size to go negative